### ISSUE TITLE
Flashing leaves huge amount of unallocated space on SD Card

### ISSUE THREAD

**Issue by ftsenguen** [cbepx-me/RG35xx-P-RG35xx-H-Modification#6]
It would be great if that unallocated space could be added to the fat32 partition. Sadly, it seems that the partition tool doesn't know how to resize the 16 and 64 mb partitions between the unallocated space and the fat32 partition, otherwise I could move things around and resize.

![image](https://github.com/cbepx-me/RG35xx-P-RG35xx-H-Modification/assets/1319741/d003fccf-9f2b-4ee1-a83b-99042cae9a0e)

**Comment by cbepx-me** [comment]
After completing the card swiping, there is no need for any other operation. Please insert the card directly and start up. The first startup will automatically complete the expansion of the game partition, create directory, and install script files,During this period, it will automatically restart. Please be patient and wait for the operation to complete

For more information, please visit:
https://github.com/cbepx-me/RG35xx-P-RG35xx-H-Modification/wiki

**Comment by ftsenguen** [comment]
Yes, the description and image are from AFTER inserting the SD card into the device and allowing it to complete set up. I was fully booting into the system before I ejected the card.

**Comment by analyticalkoala** [comment]
Hi, I was having the same problem by flashing the card using balenaEtcher on macOS.

I solved in the following way:
1) firstly formatted the card in FAT32 with GUID partition scheme;
2) flashed the mod OS by using Rufus on Windows.

This time, the OS expanded the game partition correctly.
Maybe it doesn't depend on the flashing software and the issue was just on the initial partition scheme, which on mine was MBR.

**Comment by aidehua03** [comment]
The issue does appear to be the partition mapping. My card was not properly resizing and showing a ROMS partition after the initial boot. Reformatted on a Mac to GUID partition scheme and reflashed using Balena Etcher and it works correctly now.

**Comment by cbepx-me** [comment]
According to feedback, this issue should be related to the GUID partitioning scheme. In Windows, the card swiping software automatically partitions to GUID during operation, while in macOS, there seems to be no such issue

**Comment by aidehua03** [comment]
> According to feedback, this issue should be related to the GUID partitioning scheme. In Windows, the card swiping software automatically partitions to GUID during operation, but in macOS, it seems that this is not done

That's the opposite of what it seems is happening. Balena Etcher on Windows is not setting the partition scheme to GUID on write, but leaving with whatever the card was last formatted as. The reason Mac OS has been mentioned is because it's slightly easier to switch from MBR to GUID using Disk Utility on Mac than it is on Windows (GUI vs CMD).

**Comment by James-F2** [comment]
Using the default Windows 10 formatter, now it's MBR.

![MBR](https://github.com/cbepx-me/RG35xx-P-RG35xx-H-Modification/assets/31185814/2d0e4f37-aaaf-4c6d-b44e-3ae273c0b98e)

**Comment by camplusplus** [comment]
Same for me.
Flashed with dd in linux , 
Tried with gparted and fdisk to copy then delete partition sdb1 to the end of disk , setting partition UUID to be the same but cannot fix issue and cannot boot at all.

**Comment by krei-school** [comment]
I can confirm that this is an issue (I use mac) even after setting GPT partition scheme , I flashed the card using balenaEtcher, during first boost it showed partitioning it restarted and showed  "Copying Files", however it shows the same prompt on every restart.

I checked the card using my Mac and there were no repartition done, I tried manually creating a partition for the available free space using gdisk but the gdisk didn't allow me to create the partition for the whole space, it only allowed 4.1 GB. In diskutils there is not even an option to add a partition (its disabled/greyed out).  I think the partition script included in the OS image seems failing and is messing up the partition scheme. 

I am using 64GB card

**Comment by AshokEmrys** [comment]
I can confirm that this is an issue (I use mac) even after setting GPT partition scheme , I flashed the card using balenaEtcher, during first boost it showed partitioning it restarted and showed "Copying Files", however it shows the same prompt on every restart.

I checked the card using my Mac and there were no repartition done, I tried manually creating a partition for the available free space using gdisk but the gdisk didn't allow me to create the partition for the whole space, it only allowed 4.1 GB. In diskutils there is not even an option to add a partition (its disabled/greyed out). I think the partition script included in the OS image seems failing and is messing up the partition scheme.

I am using 64GB card

**Comment by oyooyo** [comment]
I think I had the same problem: I flashed the image to a brand new 256GB Micro SD card using `dd`. When finished, I placed the card in my RG35xx H and booted.
I was greeted by a "Welcome"-screen, followed by a "Partitioning..."-screen, followed by a "Copying files..."-screen, followed by the stock main menu - at which point I shut the device down, as the initialization should be complete at this point.
But no "Roms" partition had been created on the SD card, and everytime the device booted, there was a "Copying files..."-screen before the stock main menu. I retried these steps two times, but the result stayed the same.

### What fixed the problem for me:
Immediately after writing the image to the SD card, run `fdisk` and simply write the partition tables again using the "w" command (which apparently fixes certain issues about the backup GPT table etc. that `fdisk` warns about).

**Comment by kid-scarlet** [comment]
> ### What fixed the problem for me:
> Immediately after writing the image to the SD card, run `fdisk` and simply write the partition tables again using the "w" command (which apparently fixes certain issues about the backup GPT table etc. that `fdisk` warns about).

This worked for me too! Thanks!
